addons/lunch/models/res_users.py

```diff
@@ -8,3 +8,4 @@ class ResUsers(models.Model):
     _inherit = 'res.users'
 
     last_lunch_location_id = fields.Many2one('lunch.location')
+    favorite_lunch_product_ids = fields.Many2many('lunch.product', 'lunch_product_favorite_user_rel', 'user_id', 'product_id')
```

Comment:
include favorite_lunch_product_ids in the SELF_WRITABLE fields (to prevent the modification an other user).